### ISSUE TITLE
Potential unchecked integer overflows in ticket expiration logics

### ISSUE THREAD

**Issue by briansmith** [ctz/rustls#659]
Consider:
```
            state.next_switch_time = now + u64::from(self.lifetime);
```

If `self.lifetime` is very large then the addition can overflow. If we intend for it to be wrapping then we should make that explicit. Otherwise, we should change this arithmetic to be checked.

Note that the above line is just an example. There is more unchecked arithmetic.

**Comment by ctz** [comment]
> If self.lifetime is very large then the addition can overflow.

That's true only if now is greater than 2^64 - 2^32, which is hundreds of billion years away? And even then if the user of rustls sets the ticket lifetime interval to hundreds of years. Both of these seem tremendously unlikely.

**Comment by briansmith** [comment]
> That's true only if now is greater than 2^64 - 2^32, which is hundreds of billion years away? And even then if the user of rustls sets the ticket lifetime interval to hundreds of years. Both of these seem tremendously unlikely.

Regardless, it seems like we can just use saturating addition to prevent any possibility of underflow. Note that integer overflow can result in a panic depending on the compiler configuration.
